packages/jest-circus/src/utils.js

```diff
@@ -110,46 +110,59 @@ const getEachHooksForTest = (
   return result;
 };
 
+const _makeTimeoutMessage = (timeout, isHook) => {
+  const message = `Exceeded timeout of ${timeout}ms for a ${isHook ? 'hook' : 'test'}.`;
```

Comment:
<img width="679" alt="screen shot 2017-06-08 at 9 06 43 am" src="https://user-images.githubusercontent.com/940133/26938718-e0351c6a-4c29-11e7-988f-80feee808e46.png">

timeouts are hard to deal with, but mostly because of the quality of the test.
There's not much we can do here. We can try to print the exact hook that timeouted (if it happened in a hook) or we can try to print test coverage of the test function (to see which parts weren't called), but i'm not sure how helpful it is